packages/app/src/specs/SpecsList.vue

```diff
@@ -123,15 +126,26 @@
             />
           </template>
 
+          <template #connect-button="{ utmMedium }">
+            <SpecsListCloudButton
+              v-if="row.data.isLeaf && row.data.data && (row.data.data.cloudSpec?.data || row.data.data.cloudSpec?.fetchingStatus !== 'FETCHING')"
+              :gql="props.gql"
+              @showLogin="showLogin(utmMedium)"
+              @showConnectToProject="showConnectToProject"
+              @request-access="requestAccess(props.gql?.currentProject?.projectId)"
+            />
+          </template>
+
           <template #latest-runs>
             <div
-              class="h-full grid justify-items-end items-center"
+              class="h-full grid justify-items-end items-center relative"
             >
               <RunStatusDots
                 v-if="row.data.isLeaf && row.data.data && (row.data.data.cloudSpec?.data || row.data.data.cloudSpec?.fetchingStatus !== 'FETCHING')"
                 :gql="row.data.data.cloudSpec ?? null"
                 :spec-file-extension="row.data.data.specFileExtension"
                 :spec-file-name="row.data.data.fileName"
+                @hasRuns="(v) => hasRuns = v"
```

Comment:
I found this a bit confusing - having both an event named `hasRuns` and a variable named `hasRuns`. I wonder if we can name this `setHasRuns`?

Also might be useful: `@hasRuns` compiles to `onHasRuns` - you can think of these like passing a `onHasRuns` callback in React. You can see what Vue compiles to using the [SFC Playground](https://sfc.vuejs.org/#eNpNj8FOxDAMRH/F8mVBYhPOVVrBL3D2pZSU7Yo4UeyWQ9V/x2FXiJvHnnka7/haitvWiB0GmepSFCTqWgbiJZVcFXaocYYD5poTnMx6IiaeMovCZZS3lQX65nl4fiQO/kaxvAmNqXyNGk0BhI9lg5d7pCfcoB/+ETbCIXjzmDn4vyQ+4a3IOY3FXSWzVd0bj+4HIezgd9N2VrBpwotqkc57maf24FVcrp/eJldX1iVFFyWd32v+llgNTNgQB/GBxw+mkF/r). This is why you can easily use JSX with Vue - the SFC (`.vue` file) compiles to something that basically maps 1:1 with React's JSX `h` function.

Eg: 
![image](https://user-images.githubusercontent.com/19196536/187153133-104805a0-12d3-42f1-ae10-6f03ab51a941.png)

Reply:
Removed the hasRuns by pulling up the logic into this file